src/sqlfluff/dialects/dialect_soql.py

```diff
@@ -41,11 +39,53 @@
     "NEXT_FISCAL_YEAR",
 ]
 
-soql_dialect.sets("reserved_keywords").update(date_literals)
+date_n_literals = [
```

Comment:
We use sets for other lists of date times. Any reason for straight arrays here? should we dtandardise on sets to match other dialects?